File: ClientBridge/src/main/php/client/enums/ClientType.php
<?php
/**
 * Created by Nick on 21 okt. 2019.
 * Copyright © ImSpooks
 */

namespace client\enums;


class ClientType {
    public const JAVA = ["ID" => 0, "NAME" => "JAVA"];
    public const PHP = ["ID" => 1, "NAME" => "PHP"];

    private static $CACHE = [self::JAVA, self::PHP];

    public static function getFromName(string $name): array {
        foreach (self::$CACHE as $vars) {
            if (strcasecmp($vars["NAME"], $name)) {
                return $vars;
            }
        }
        return [];
    }


    public static function getFromId(int $id): array {
        foreach (self::$CACHE as $vars) {
            if (strcasecmp($vars["ID"], $id)) {
                return $vars;
            }
        }
        return [];
    }
}

File: Resources/src/main/java/me/ImSpooks/core/enums/ClientType.java
package me.ImSpooks.core.enums;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

/**
 * Created by Nick on 21 okt. 2019.
 * Copyright © ImSpooks
 */
@RequiredArgsConstructor
public enum ClientType {
    NULL(-1, ""),
    SERVER(-1, "SERVER"),
    JAVA(0, "JAVA"),
    PHP(1, "PHP"),
    ;

    @Getter private final int id;
    @Getter private final String name;

    public static ClientType[] CACHE = values();

    public static ClientType getFromName(String name) {
        for (ClientType clientType : CACHE) {
            if (clientType.getName().equalsIgnoreCase(name)) {
                return clientType;
            }
        }
        return NULL;
    }

    public static ClientType getFromId(int id) {
        for (ClientType clientType : CACHE) {
            if (clientType.getId() == id) {
                return clientType;
            }
        }
        return NULL;
    }
}
